spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -154,6 +172,17 @@ private boolean supportsParquetBatchReads(Types.NestedField field) {
     return field.type().isPrimitiveType() || MetadataColumns.isMetadataColumn(field.fieldId());
   }
 
+  private boolean useCometBatchReads() {
+    return readConf.parquetVectorizationEnabled()
+        && readConf.parquetReaderType() == ParquetReaderType.COMET
+        && expectedSchema.columns().stream().allMatch(this::supportsCometBatchReads)
+        && taskGroups.stream().allMatch(this::supportsParquetBatchReads);
```

Comment:
It is safe to reuse `supportsParquetBatchReads` because the logic is the same?

Reply:
Yes, the logic is the same.